torchgeo/trainers/chesapeake.py

```diff
@@ -296,90 +294,138 @@ def __init__(
         self.train_state = train_state
         self.layers = ["naip-new", "lc"]
         self.patches_per_tile = patches_per_tile
-        self.original_patch_size = 500
+        self.patch_size = patch_size
+        # This is a rough estimate of how large of a patch we will need to sample in
+        # EPSG:3857 in order to garuntee a large enough patch in the local CRS.
+        self.original_patch_size = int(patch_size * 2.0)
         self.batch_size = batch_size
         self.num_workers = num_workers
 
-    def custom_transform(
-        self, sample: Dict[str, Any], patch_size: int = 256
-    ) -> Dict[str, Any]:
-        """Transform a single sample from the Dataset.
-
-        Args:
-            sample: a single sample from the Dataset
-            patch_size: size of center cropped patch to return
-
-        Returns:
-            a transformed sample
-        """
-        # Center crop
-        num_image_channels, height, width = sample["image"].shape
-        num_mask_channels = sample["mask"].shape[0]
-
-        # If we somehow sample a patch that is smaller than the `patch_size` we want to
-        # sample, then we create a nodata patch instead
-        if height < patch_size or width < patch_size:
-            height, width = patch_size, patch_size
-            sample["image"] = torch.zeros(  # type: ignore[attr-defined]
-                (num_image_channels, patch_size, patch_size)
+    def pad_to(
```

Comment:
At one point in the code there was a TODO suggesting that we should refactor this to use normal transforms. Now that we're using more and more transforms in here, we may want to revisit this.

Reply:
I considered this refactored in the sense that there isn't a giant `do_all_the_work(img)` function that is applied to the train/val/test Datasets. When we get the Kornia transforms working with our data dicts then this will need to be revisited.

Reply:
(and at that point, center crop and pad to would be handled by Kornia transforms and removed from this datamodule)